File: src/main/java/com/undsf/mcmodmgr/curseforge/models/PaginationResponse.java
package com.undsf.mcmodmgr.curseforge.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class PaginationResponse<D> {
    @JsonProperty
    public List<D> data;

    @JsonProperty
    public Pagination pagination;
}

File: src/main/kotlin/com/undsf/mcmodmgr/curseforge/responses/PaginationResponse.kt
package com.undsf.mcmodmgr.curseforge.responses

class PaginationResponse<D>(data: List<D>, var pagination: Pagination) : DataResponse<List<D>>(data) {
    override fun toString(): String {
        return "${pagination.index} ~ ${pagination.index + pagination.resultCount} of ${pagination.totalCount}"
    }
}